### PR TITLE
Fix ParallelizableAttribute with TestFixtureSource

### LINKED ISSUE #3735: Parallelizable TestFixtureSource with TestFixtureData not running in parallel
NUnit (targeting .NET 5) does not run my TestFixture in parallel instances when using TestFixtureSource:

```
public class TestFixtureCases
{
	public static IEnumerable Cases
	{
		get
		{
			yield return new TestFixtureData("string1");
			yield return new TestFixtureData("string2");
			yield return new TestFixtureData("string3");
		}
	}
}
[TestFixture]
[Parallelizable]
[TestFixtureSource(typeof(TestFixtureCases), nameof(TestFixtureCases.Cases))]
public class FixtureClass : BasicFixtureClass
{
	public FixtureClass(string fixtureCase) : base(fixtureCase)
	{ }

	[Test]
	public void test1()
	{ }
	[Test]
	public void test2()
	{ }
}
```

However, if I'm giving the FixtureCases like the below, it _does_ run in multiple instances:
```
[TestFixture("string1")]
[TestFixture("string2")]
[TestFixture("string3")]
[Parallelizable]
public class FixtureClass : BasicFixtureClass
{
	public FixtureClass(string fixtureCase) : base(fixtureCase)
	{ }

	[Test]
	public void test1()
	{ }
	[Test]
	public void test2()
	{ }
}
```

### PR DESCRIPTION
**Pull request by jairbubbles**
Tests declared like that:

```csharp
[TestFixtureSource("Data")]
[Parallelizable(ParallelScope.All)]
public class TestFixtureSourceWithParallelizableAttribute
```

were no longer run in parallel since that change: https://github.com/nunit/nunit/commit/247ede8aab1f75a99218bcabbe1eae4692b99215

Similar to #3715 I explicitely apply `ParallelizableAttribute` to children tests.

Closes #3735

NB: I couldn't run tests in master branch so I tested in the 3.0 dev branch.

### CODE REVIEW

**Comment by jontdelorme** [comment]
Hello, we are waiting for that fix to update to the latest version. Anyone ready for the review?

**Comment by jairbubbles** [comment]
Hey @SimonCropp anything I can do to get this merged?

**Comment by SimonCropp** [comment]
@jairbubbles LGTM. but i am not a maintainer on this project, so cant merge

**Comment by jairbubbles** [comment]
Oh sorry I found your name in the last PRs merged but I guess you were just the author then.

**Comment by jairbubbles** [comment]
(Thx @stevenaw for updating my branch)

**Comment by jairbubbles** [comment]
Hi @stevenaw, anything I can do to help this PR getting merged?

**Comment by stevenaw** [comment]
Hi @jairbubbles thanks for asking. Nothing on your end, this just needs a review by the team. Sorry it's taken so long to be looked at. I'd hoped to review it but I got a little unexpectedly busy myself. I still hope to get to this soon.

**Comment by OsirisTerje** [comment]
@stevenaw This looks very simple.  Do you think we can just merge it?

**Review by stevenaw — APPROVED** [review]
I'd be alright with that @OsirisTerje 
This LGTM too, thanks @jairbubbles for your contribution!

**Comment by stevenaw** [comment]
I'm having a bit of a hard time merging.

**Comment by jairbubbles** [comment]
I rebased and resolved the conflict manually.

**Review by SimonCropp — APPROVED** [review]

**Pull request merged**
